.github/workflows/update-dfx.yml

```diff
@@ -0,0 +1,64 @@
+# A GitHub Actions workflow that regularly checks for a new dfx
+# and creates a PR on new versions.
+name: Update dfx
+on:
+  schedule:
+    # check for new dfx versions weekly
+    - cron: '30 3 * * FRI'
+  workflow_dispatch:
+  # Provide an option to run this manually.
+  push:
+    branches:
+      # Development branch for this workflow:
+      - "update-dfx"
+jobs:
+  update-dfx:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        # First, check dfx GitHub releases for a new version. We assume that the
+        # latest version's tag name is the version.
+      - name: Check new dfx version
+        id: update
+        run: |
+          current_dfx_version="$(jq -r .dfx dfx.json)"
+          echo "Current dfx version: '$current_dfx_version'"
+          latest_dfx_version="$(curl -sSL https://sdk.dfinity.org/manifest.json | jq -r .tags.latest)"
+          echo "Latest dfx version: '$latest_dfx_version'"
+
+          if [ "$current_dfx_version" != "$latest_dfx_version" ]
```

Comment:
If I recall correctly, we started using version `0.13.0` in nns-dapp before it was officially marked as the latest version. Should we check for `-lt` instead of `!=`?

Reply:
I would not trust bash lt to compare semantic version numbers out of the box.

I believe that exceptional circumstances, such as using `0.14.1` when `0.14.0` is still marked as latest is best dealt with manually, by rejecting the PR.